src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs

```diff
@@ -63,7 +63,8 @@ public Task<LanguageServer.Protocol.CompletionItem[]> HandleAsync(object input,
             // However, this works through liveshare on the LSP client, but not the LSP extension.
             // see https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1107682 for tracking.
             var request = ((JObject)input).ToObject<CompletionParams>(s_jsonSerializer);
-            var context = new LSP.RequestContext(requestContext.GetClientCapabilities(), null);
+            var textDocument = GetTextDocumentIdentifier(request);
+            var context = SolutionProvider.CreateRequestContext(textDocument, requestContext.GetClientCapabilities());
             return base.HandleRequestAsync(request, context, cancellationToken);
```

Comment:
This is where the API falls down a little. I don't love that these have to know to call `GetTextDocumentIndentifier` because handling the request, but at the same time this `TypeScriptHandlerShims` is a bit of an ugly duckling anyway. Open to suggestions

Reply:
Sadly, no. Each one inherits from the "real" handler that does the work. I could simply this all into the `CreateRequestContext` method, now that its in this file, if you think that will be useful?

Reply:
Yeah that would be good, unify the logic a little bit.